src/lib/device-types/index.ts

```diff
@@ -187,6 +196,29 @@ function updateDeviceTypesCache(
 	});
 }
 
+export function syncDataModel(
+	types: Dictionary<DeviceTypeInfo>,
+	propertyMap: Dictionary<string>,
+) {
+	return db.transaction(tx => {
+		return Promise.each(Object.values(types), deviceTypeInfo => {
+			const deviceType = deviceTypeInfo.latest.deviceType;
+			const body = _.mapValues(
+				_.invert(propertyMap),
+				source => (deviceType as AnyObject)[source],
+			);
+			return updateOrInsertModel(
```

Comment:
If we keep calling this on each run of `fetchDeviceTypesAndReschedule` would it make sense to change `updateOrInsert` to only run an UPDATE if the objects are different?

Reply:
I would not change `updateOrInsert` but maybe add a new function that behaves the way you describe, so that we don't break anything.